graph/transaction.go

```diff
@@ -52,14 +48,41 @@ func (t *Transaction) AddQuad(q quad.Quad) {
 // The quad will be removed from the database if it is not present in the
 // transaction, otherwise it simply remove it from the transaction.
 func (t *Transaction) RemoveQuad(q quad.Quad) {
-	ad := Delta{
+	ad, rd := createDeltas(q)
+
+	if _, adExists := t.deltas[ad]; adExists {
+		t.deleteDelta(ad)
+	} else {
+		if _, rdExists := t.deltas[rd]; !rdExists {
+			t.addDelta(Delta{Quad: q, Action: Delete})
+		}
+	}
+}
+
+func createDeltas(q quad.Quad) (ad, rd Delta) {
+	ad = Delta{
 		Quad:   q,
 		Action: Add,
 	}
+	rd = Delta{
+		Quad:   q,
+		Action: Delete,
+	}
+	return
+}
 
-	if _, adExists := t.Deltas[ad]; adExists {
-		delete(t.Deltas, ad)
-	} else {
-		t.Deltas[Delta{Quad: q, Action: Delete}] = struct{}{}
+func (t *Transaction) addDelta(d Delta) {
+	t.Deltas = append(t.Deltas, d)
+	t.deltas[d] = struct{}{}
+}
+
+func (t *Transaction) deleteDelta(d Delta) {
```

Comment:
But that's a bad idea because updating everything would be a pain. As-is is good.